### PR TITLE
OCPBUGS-39333:  E2E: Add support for hypershift to  ovs dynamic pinning and kubelet.experimental annotation tests

### PR DESCRIPTION
**Pull request by mrniranjan**
- use appropriate resource pool  nodepools for hypershift and mcp for regular cluster 
-  replace dataplane test client instead of generic testclient.Client

### CODE REVIEW

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go [hunk 1]
```diff
@@ -74,18 +76,24 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, Label(string(lab
 		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
 		Expect(err).ToNot(HaveOccurred())
 
-		performanceMCP, err = mcps.GetByProfile(profile)
-		Expect(err).ToNot(HaveOccurred())
+		if !hypershift.IsHypershiftCluster() {
+			poolName, err = mcps.GetByProfile(profile)
+			Expect(err).ToNot(HaveOccurred())
+		} else {
+			hostedClusterName, err := hypershift.GetHostedClusterName()
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:83]: We need to check for the error here

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go [hunk 2]
```diff
@@ -74,18 +76,24 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, Label(string(lab
 		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
 		Expect(err).ToNot(HaveOccurred())
 
-		performanceMCP, err = mcps.GetByProfile(profile)
-		Expect(err).ToNot(HaveOccurred())
+		if !hypershift.IsHypershiftCluster() {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:79]: I would suggest to add
`GetPoolName` function that would handle the case for both OCP and HCP

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/kubelet.go [hunk 3]
```diff
@@ -45,33 +52,38 @@ var _ = Describe("[ref_id: 45487][performance]additional kubelet arguments", Ord
 		profile, err = profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
 		Expect(err).ToNot(HaveOccurred())
 
-		performanceMCP, err = mcps.GetByProfile(profile)
-		Expect(err).ToNot(HaveOccurred())
+		if !hypershift.IsHypershiftCluster() {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/7_performance_kubelet_node/kubelet.go:55]: You can use a `GetPoolName` function hide the details about the platform and handle both cases

**Review by Tal-or — COMMENTED** [review]
Thank you for working on this, left couple of comments

**Comment by Tal-or** [comment]
```
  STEP: Reverting the Profile - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:188 @ 10/25/24 08:15:12.157
  STEP: Applying changes in performance profile and waiting until clusters/24a18b9e1fced5fb7ddc-us-east-1a will start updating - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:196 @ 10/25/24 08:15:12.198
  Oct 25 08:15:12.201: [INFO]: wait for node pool "clusters/24a18b9e1fced5fb7ddc-us-east-1a" transition into update config state
  STEP: Waiting when clusters/24a18b9e1fced5fb7ddc-us-east-1a finishes updates - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:199 @ 10/25/24 08:15:22.209
  Oct 25 08:15:22.212: [INFO]: wait for node pool "clusters/24a18b9e1fced5fb7ddc-us-east-1a" transition into config ready state
  [FAILED] Unexpected error:
      <*url.Error | 0xc0006a2d50>: 
      Get "https://a9c92984710fa4b81a716a01d0bad186-164ea1e233c0eee6.elb.us-east-1.amazonaws.com:6443/apis/hypershift.openshift.io/v1beta1/namespaces/clusters/nodepools/24a18b9e1fced5fb7ddc-us-east-1a": http2: client connection lost
      {
          Op: "Get",
          URL: "https://a9c92984710fa4b81a716a01d0bad186-164ea1e233c0eee6.elb.us-east-1.amazonaws.com:6443/apis/hypershift.openshift.io/v1beta1/namespaces/clusters/nodepools/24a18b9e1fced5fb7ddc-us-east-1a",
          Err: <*errors.errorString | 0xc00029b140>{
              s: "http2: client connection lost",
          },
      }
  occurred
  In [AfterEach] at: /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:168 @ 10/25/24 08:43:27.24
  < Exit [AfterEach] [Performance Profile Modified] - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:187 @ 10/25/24 08:43:27.24 (28m15.083s)
  Attempt #1 Failed.  Retrying ↺ @ 10/25/24 08:43:27.244
  > Enter [BeforeEach] [performance] Cgroups and affinity - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:97 @ 10/25/24 08:43:27.244
  STEP: Checking the profile performance with cpus reserved=[0] isolated=[1-3] - /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:98 @ 10/25/24 08:43:27.245
  Oct 25 08:43:47.272: [INFO]: daemonset "node-inspector-ns" "node-inspector" desired 3 scheduled 3 ready 3
  [FAILED] Unexpected error:
      <*errors.errorString | 0xc000694c70>: 
      failed to get daemon pod for the node "ip-10-0-130-98.ec2.internal"
      {
          s: "failed to get daemon pod for the node \"ip-10-0-130-98.ec2.internal\"",
      }
  occurred
  In [BeforeEach] at: /go/src/github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go:104 @ 10/25/24 08:43:47.275
```

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ import (
 var _ = BeforeSuite(func() {
 	Expect(testclient.ClientsEnabled).To(BeTrue())
 	// create test namespace
-	err := testclient.Client.Create(context.TODO(), namespaces.TestingNamespace)
+	err := testclient.ControlPlaneClient.Create(context.TODO(), namespaces.TestingNamespace)
```
Comment (rbaturov) [inline test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go:34]: This  should be `DataPlaneClient`

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go [hunk 5]
```diff
@@ -40,7 +40,7 @@ var _ = BeforeSuite(func() {
 })
 
 var _ = AfterSuite(func() {
-	err := testclient.Client.Delete(context.TODO(), namespaces.TestingNamespace)
+	err := testclient.ControlPlaneClient.Delete(context.TODO(), namespaces.TestingNamespace)
```
Comment (rbaturov) [inline test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go:43]: Here as well ^

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go (hunk 4 above)
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go]: fixed in the latest commit

File: test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go (hunk 5 above)
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/7_performance_kubelet_node/test_suite_performance_kubelet_node_test.go]: fixed in the latest commit

**Pull request merged**